jupyter_collaboration/loaders.py

```diff
@@ -0,0 +1,108 @@
+from __future__ import annotations
+
+import asyncio
+from logging import Logger, getLogger
+from typing import Any, Callable, Coroutine
+
+from jupyter_server.utils import ensure_async
+
+
+class FileLoader:
+    def __init__(
+        self,
+        file_id: str,
+        file_format: str,
+        file_type: str,
+        file_id_manager: Any,
+        contents_manager: Any,
+        log: Logger | None,
+        poll_interval: float | None = None,
```

Comment:
Seems that this was introduced only in 3.10, could you keep using `Optional` for now:

```suggestion
        log: Optional[Logger],
        poll_interval: Optional[float]= None,
```

Reply:
You can use it with older Python versions if you add `from __future__ import annotations`.